packages/flutter_web_plugins/lib/src/navigation/js_url_strategy.dart

```diff
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+// ignore_for_file: library_private_types_in_public_api
```

Comment:
@mdebbar Can you help me with this file?

The types `_JsSetUrlStrategy`, `_PathGetter`, `_StateGetter`, `_AddPopStateListener`, `_StringToString`, `_StateOperation`, `_HistoryMove` are declared private, but they are used in public API which - among other things - is strange in docs ([1], [2]).

We could either make the public places that use the private types in this file private (probably not the right move) or make these types public and then document them properly. Unfortunately, I don't know what the documentation for these should be.

[1] https://master-api.flutter.dev/flutter/flutter_web_plugins/JsUrlStrategy/JsUrlStrategy.html
[2] https://master-api.flutter.dev/flutter/flutter_web_plugins/jsSetUrlStrategy.html

Reply:
I've left this in as a TODO for now to get the other parts of this PR landed.